File: src/main/java/com/vjiki/music/dto/PlaylistSongResponse.java
package com.vjiki.music.dto;

import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PlaylistSongResponse {
    private UUID id;
    private UUID playlistId;
    private UUID songId;
    private String songTitle;
    private String songArtist;
    private String songAudioUrl;
    private String songCoverUrl;

    @Builder.Default
    private List<TagResponse> tags = new ArrayList<>();

    private Integer position;
    private OffsetDateTime addedAt;
    private UUID addedBy;
}

File: src/main/kotlin/com/vjiki/music/dto/PlaylistSongResponse.kt
package com.vjiki.music.dto

import java.time.OffsetDateTime
import java.util.UUID

data class PlaylistSongResponse(
    val id: UUID,
    val playlistId: UUID,
    val songId: UUID,
    val songTitle: String?,
    val songArtist: String?,
    val songAudioUrl: String?,
    val songCoverUrl: String?,
    val position: Int,
    val addedAt: OffsetDateTime?,
    val addedBy: UUID?
)

